src/internal_events/tcp.rs

```diff
@@ -190,3 +190,70 @@ impl InternalEvent for TcpBytesReceived {
         );
     }
 }
+
+pub struct TcpListenerError<'a, E> {
+    pub error: E,
+    pub message: &'a str,
+}
+
+impl<E: std::fmt::Display> InternalEvent for TcpListenerError<'_, E> {
+    fn emit(self) {
+        error!(
+            message = %self.message,
+            error = %self.error,
+            error_type = error_type::CONFIGURATION_FAILED,
+            stage = error_stage::RECEIVING,
+            internal_log_rate_secs = 10,
+        );
+    }
```

Comment:
Per offline discussion-

If we are intentionally not incrementing `component_errors_total` ,
 seems like this should stay  an inline error! call . To be an `<...>Error` internal event, it should be incrementing a counter.

Reply:
Reverted to the original `error!` calls in https://github.com/vectordotdev/vector/pull/14422/commits/2bfdd251da4cc2eaf9b66e7fcfda9782fb47545a